Bukkit/src/main/java/com/plotsquared/bukkit/listeners/PlayerEvents.java

```diff
@@ -1263,7 +1263,8 @@ public void onBlockPistonRetract(BlockPistonRetractEvent event) {
     @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
     public void onBlockDispense(BlockDispenseEvent event) {
         Material type = event.getItem().getType();
-        if (type != Material.WATER_BUCKET && type != Material.LAVA_BUCKET) {
+        Material dispenserType = event.getBlock().getType();
```

Comment:
So, great that this is fixed (I assume it's a bug fix as well?). 

However, for future reference I'd suggest that things like this would be a separate pull request. I'll let this one pass, because it's simple enough. :)

Reply:
yea sorry for that :)